doc/contributing/contributing-to-documentation.chapter.md

```diff
@@ -120,3 +120,34 @@ Additional syntax extensions are available, though not all extensions can be use
   > :   green fruit with red flesh
 
 For contributing to the legacy parts, please see [DocBook: The Definitive Guide](https://tdg.docbook.org/) or the [DocBook rocks! primer](https://web.archive.org/web/20200816233747/https://docbook.rocks/).
+
+## Diagrams
+
+Diagrams are provided to contextualize the amount of information and processes
+into something more visual.
+
+The diagrams are generated using [Plantuml][plantuml website] and the sources
+are available in this repository, in `doc/diagrams`. Find plethora of examples
+and documentation also on the [Plantuml][plantuml website].
+
+The procedure to submit changes for these diagrams follow the usual process with
+Github pull-requests including the relevant changes except that pull-requests
+must include a link to the updated diagram using the web interface from
+[Plantuml][plantuml web interface website] or [Planttext][planttext website] so
+we can visually review the new changes.
+
+To render diagrams locally use
+
+```shell
+nix-shell '<nixpkgs>' --pure -p plantuml --run "plantuml path/to/diagram.plantuml"
+```
+
+Or if you're using flake
+
+```
+nix run nixpkgs#plantuml path/to/diagram.plantuml
+```
+
+[plantuml website]: https://plantuml.com/
+[plantuml web interface website]: https://www.plantuml.com/plantuml/
+[planttext website]: https://planttext.com/
```

Comment:
- you can use implicit reference links for better readability and easier maintainance
- link to source files for easier disovery
- put reference links close to where they are used
- GitHub calls it "pull request", therefore so do we
- use short sentences where possible
- it seems, the Nixpkgs manual does not document flake-based workflows. let's keep it consistent then.
- since there is no automatic process, we need a note that it has to be taken care of manually.
- present command line examples such that it is absolutely clear how to change them to actually work, or in a way that no change is neccessary

````suggestion
The diagrams in this manual are generated using [Plantuml].
Their sources are available in [`/doc/diagrams`][diagrams].

Currently there is automated process to render images from those sources and include them in the manual.
Therefore, rendered images have to be added and updated manually.

When making a pull request to change a diagram, to ease review, please provide a link to a rendering of the updated diagram in the pull request description.
Generate a link with the [Plantuml web interface] or [Planttext].

[diagrams]: https://github.com/NixOS/nixpkgs/tree/master/doc/diagrams
[Plantuml]: https://plantuml.com/
[Plantuml web interface]: https://www.plantuml.com/plantuml/
[Planttext]: https://planttext.com/

To render diagrams locally, run the following command in your copy of the `nixpkgs` Git repository:

```shell
nix-shell '<nixpkgs>' --pure -p plantuml --run "plantuml doc/diagrams/*.plantuml"
```
````